src/include/baselib/jni/JniEnvironment.h

```diff
@@ -136,6 +137,61 @@ namespace bl
                 return m_jniEnv -> GetVersion();
             }
 
+            jobjectRefType getObjectRefType( SAA_in const jobject object ) const NOEXCEPT
+            {
+                return m_jniEnv -> GetObjectRefType( object );
+            }
+
+            void deleteLocalRef( SAA_in const jobject object) const NOEXCEPT
+            {
+                m_jniEnv -> DeleteLocalRef( object );
+            }
+
+            jobject newGlobalRef( SAA_in const jobject object ) const
+            {
+                const auto globalRef = m_jniEnv -> NewGlobalRef( object );
+
+                BL_CHK_T(
+                    nullptr,
+                    globalRef,
+                    JavaException(),
+                    BL_MSG()
+                        << "Failed to create new global reference."
+                    );
+
+                return globalRef;
+            }
+
+            void deleteGlobalRef( SAA_in const jobject object) const NOEXCEPT
+            {
+                m_jniEnv -> DeleteGlobalRef( object );
+            }
+
+            LocalReference< jclass > findJavaClass( SAA_in const std::string& javaClassName ) const
+            {
+                const auto javaClass = m_jniEnv -> FindClass( javaClassName.c_str() );
```

Comment:
you need to immediately wrap this into local reference wrapper to make sure the code is always exception safe (no matter how it changes in the future)